sdk/python/kfp/_client.py

```diff
@@ -139,6 +144,16 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client
     self._experiment_api = kfp_server_api.api.experiment_service_api.ExperimentServiceApi(api_client)
     self._pipelines_api = kfp_server_api.api.pipeline_service_api.PipelineServiceApi(api_client)
     self._upload_api = kfp_server_api.api.PipelineUploadServiceApi(api_client)
+    self._api_client = api_client
+    if self._context_setting['namespace'] == '' and self.get_kfp_healthz().get('multi_user') is True:
+      NAMESPACE_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+      try:
+        with open(NAMESPACE_PATH, 'r') as f:
+          current_namespace = f.read()
+          self.set_user_namespace(current_namespace)
+      except:
+        print('Failed to automatically set namespace.')
```

Comment:
nit: Consider using proper logging module, e.g., `absl.logging` where users can control the verbosity and the output stream of logging info.